include/highfive/bits/H5Easy_vector.hpp

```diff
@@ -53,6 +53,23 @@ struct overwrite_impl
         return dataset;
     }
 };
+    
+// load "std::vector<std::vector>" from DataSet
+template <class T>
+struct load_impl<std::vector<std::vector<T>>>
+{
+    static std::vector<std::vector<T>> run(const File& file, const std::string& path)
+    {
+        DataSet dataset = file.getDataSet(path);
+        std::vector<size_t> dims = dataset.getDimensions();
+        if (dims.size() != 2) {
+            throw detail::error(file, path, "H5Easy::load: Field not rank 2");
+        }
+        std::vector<std::vector<T>> data;
+        dataset.read(data);
+        return data;
+    }
+};
 
 }  // namespace vector
```

Comment:
The reason your code is failing is that you are addition your overload to the `H5Easy::detail::vector` namespace, whereas the overloads are selected from the `H5Easy::detail` namespace. Practically this means that you need to reposition this closing bracket (`}  // namespace vector`) above your added code.

Reply:
A side note is that I do think it is important to keep this overload before the `std::vector<T>` overload (as you already did).

Reply:
I'm fine with any of the suggested changes, as long as I can load vector<vector<T>>. Removing the `if (dims.size() != 1) {` seems like the simplest solution to me. But I didn't want to remove it as I thought it might be there for some reason. But possibly the author just didn't consider that T may be a vector itself and everything still works out.

Reply:
Since I am the author I can confirm ;). I would suggest that `if (dims.size() != 1) {` be replaced by `if (dims.size() != detail::vector::getDims(data)) {`. Whereby:

* The check should be placed after the allocation of `data`,
* `getDims(data)` should be on the lines of [this answer](https://stackoverflow.com/a/28942971/2646505).